### DISCUSSION TITLE
consensus/raft-log-replication

### DISCUSSION THREAD

**Discussion by giscus** [isno/theByteBook#128] [General]
# consensus/raft-log-replication

构建大规模高可用的分布式系统

https://www.thebyte.com.cn/consensus/raft-log-replication.html

**Comment by littlebear-xbz** [comment]
时 应该为 是

# 日志复制
raft 是强 leader 模型的算法，日志项只能由 leader 复制给其他成员，这意味着日志复制**时**单向的

**Reply by isno** [reply]
谢谢，已修复

**Reply by JackCharles** [reply]
“Leader 处理不一致的具体过程分析如下”
这一部分第2和第3步，感觉缺了一点东西：
应该是Follower拒绝日志后，Leader会递减该Follower的LogIndex（nextIndex）,然后尝试发送索引值为 6，任期编号为 3 的日志项。
完了才是第3点，接受日志，找到一致的最大索引。

**Comment by xierunxiang55** [comment]
跟随者收到志复制消息后 -> 跟随者收到日志复制消息后

**Reply by isno** [reply]
谢谢，已修正。

**Comment by yinyangsx** [comment]
一旦选出一个公认的领导者，那领导者顺理其章地承担起“处理系统发生的所有变更，并将变更复制到所有跟随者节点”的职责。
这句话中有个错别字 “顺理其章“中的“其”，应该改成“成” -- 顺理成章

**Reply by isno** [reply]
感谢，这本书质量又提高啦！
